src/pykeen/sampling/bernoulli_negative_sampler.py

```diff
@@ -96,14 +98,26 @@ def sample(self, positive_batch: torch.LongTensor) -> torch.LongTensor:
             device=positive_batch.device,
         )
 
-        # Replace heads – To make sure we don't replace the head by the original value
-        # we shift all values greater or equal than the original value by one up
-        # for that reason we choose the random value from [0, num_entities -1]
-        filter_same_head = (negative_entities[head_mask] >= positive_batch[:, 0][head_mask])
-        negative_batch[:, 0][head_mask] = negative_entities[head_mask] + filter_same_head.long()
+        # Replace heads
+        negative_batch[:, 0][head_mask] = negative_entities[head_mask]
 
         # Replace tails
-        filter_same_tail = (negative_entities[tail_mask] >= positive_batch[:, 2][tail_mask])
-        negative_batch[:, 2][tail_mask] = negative_entities[tail_mask] + filter_same_tail.long()
-
-        return negative_batch
+        negative_batch[:, 2][tail_mask] = negative_entities[tail_mask]
+
+        # If filtering is activated, all negative triples that are positive in the training dataset will be removed
+        if self.filtered:
+            batch_filter = self._filter_negative_triples(negative_batch=negative_batch)
+            negative_batch = negative_batch[batch_filter]
+        else:
+            # To make sure we don't replace the head by the original value
+            # we shift all values greater or equal than the original value by one up
+            # for that reason we choose the random value from [0, num_entities -1]
+            negative_batch[:, 0][head_mask] += (
```

Comment:
is `negative_batch[:, 0][head_mask] ` the same as `negative_batch[head_mask, 0]`?

Reply:
Yes. Well spotted 👍 I just used the old code 😅